tests/StatsdClient.Tests/StatsdConfigurationTests.cs

```diff
@@ -11,65 +11,91 @@ namespace Tests
             Thread listenThread = new Thread(new ParameterizedThreadStart(udpListener.Listen));
             listenThread.Start();
-            DogStatsd.Increment(testCounterName);
+            dogStatsdService.Increment(testCounterName);
             while (listenThread.IsAlive) ;
             Assert.AreEqual(expectedOutput, udpListener.GetAndClearLastMessages()[0]);
             udpListener.Dispose();
         }
 
+        private DogStatsdService CreateSut()
+        {
+            return new DogStatsdService();
+        }
+
         [Test]
         public void throw_exception_when_no_config_provided()
         {
+            var sut = CreateSut();
             StatsdConfig metricsConfig = null;
-            Assert.Throws<ArgumentNullException>(() => StatsdClient.DogStatsd.Configure(metricsConfig));
+            Assert.Throws<ArgumentNullException>(() => sut.Configure(metricsConfig));
         }
 
         [Test]
         public void throw_exception_when_no_hostname_provided()
         {
+            var sut = CreateSut();
             var metricsConfig = new StatsdConfig { };
-            Assert.Throws<ArgumentNullException>(() => StatsdClient.DogStatsd.Configure(metricsConfig));
+            Assert.Throws<ArgumentNullException>(() => sut.Configure(metricsConfig));
         }
 
         [Test]
         public void default_port_is_8125()
         {
+            var sut = CreateSut();
             var metricsConfig = new StatsdConfig
             {
                 StatsdServerName = "127.0.0.1"
             };
-            StatsdClient.DogStatsd.Configure(metricsConfig);
-            testReceive("127.0.0.1", 8125, "test", "test:1|c");
+            sut.Configure(metricsConfig);
+            testReceive("127.0.0.1", 8125, "test", "test:1|c", sut);
         }
 
         [Test]
         public void setting_port()
         {
+            var sut = CreateSut();
             var metricsConfig = new StatsdConfig
             {
                 StatsdServerName = "127.0.0.1",
                 StatsdPort = 8126
             };
-            StatsdClient.DogStatsd.Configure(metricsConfig);
-            testReceive("127.0.0.1", 8126, "test", "test:1|c");
+            sut.Configure(metricsConfig);
+            testReceive("127.0.0.1", 8126, "test", "test:1|c", sut);
         }
 
         [Test]
         public void setting_prefix()
         {
+            var sut = CreateSut();
             var metricsConfig = new StatsdConfig
             {
                 StatsdServerName = "127.0.0.1",
                 StatsdPort = 8129,
                 Prefix = "prefix"
             };
-            StatsdClient.DogStatsd.Configure(metricsConfig);
-            testReceive("127.0.0.1", 8129, "test", "prefix.test:1|c");
+            sut.Configure(metricsConfig);
+            testReceive("127.0.0.1", 8129, "test", "prefix.test:1|c", sut);
+        }
+
+        [Test]
+        public void ServiceCannotBeConfiguredMoreThanOnce()
```

Comment:
nitpick: for consistency with other tests, I'd make the method name camel_case (even though it may not follow the language convention).

Reply:
What about if I rename all the other tests to be .NET standard, since this is C# after all, is that an option? If not and you really want to keep the snake casing (camel casing is something different) then I'll change it

Reply:
My bad i meant snake_case-ing indeed. 

I don't have anything against renaming all other tests, I just figured it'd be faster to modify this one only.